sky/utils/controller_utils.py

```diff
@@ -661,6 +661,65 @@ def replace_skypilot_config_path_in_file_mounts(
                      f'with the real path in file mounts: {file_mounts}')
 
 
+def _generate_run_uuid() -> str:
+    """Generates a unique run id for the job."""
+    return common_utils.base36_encode(uuid.uuid4().hex)[:8]
+
+
+def translate_local_file_mounts_to_two_hop(
+        task: 'task_lib.Task') -> Dict[str, str]:
+    """Translates local->VM mounts into two-hop file mounts.
+
+    This strategy will upload the local files to the controller first, using
+    a normal rsync as part of sky.launch(). Then, when the controller launches
+    the task, it will also use local file_mounts from the destination path of
+    the first hop.
```

Comment:
```suggestion
    This strategy will upload the local files to the controller first, using
    a normal rsync as part of sky.launch() for the controller. Then, when the
    controller launches the task, it will also use local file_mounts from the
    destination path of the first hop.
```